### PR TITLE
Databond support

### LINKED ISSUE #685: Databond support
# Rationale

Incase a block is not confirmed in the previous epoch, razor go would need to do offchain calculations on occurrence for all data bonds and check whether they need to be reported upon in this epoch.

# Implementation

### PR DESCRIPTION
**Pull request by shekhar2807**
# Description

- Added data-bond functions in commit.go
- Maintained registry from leafIdToCollectionId
- Create collection modified as occurrence flag is added
- Create job now uses createMulJobs contract function which takes array of jobs as input
- Tests are modified according to the above changes

Fixes #685 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Contracts version:
* Hardware:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/commit_test.go [hunk 1]
```diff
@@ -380,6 +463,224 @@ func TestGetSalt(t *testing.T) {
...
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    uint16
+		want1   []bindings.StructsCollection
+		wantErr bool
+	}{
+		{
+			name: "Test 1: When ModifyCollections() executes successfully",
```
Comment (ashish10677) [inline cmd/commit_test.go:588]: Add more test cases with better values to cover as many edge cases as possible.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

**Comment by SkandaBhat** [comment]
Plesae open this once this has been picked up again.
